android/src/main/java/com/optimove/flutter/OptimoveFlutterPlugin.java

```diff
@@ -293,6 +301,34 @@ private void getInboxItems(@NonNull Result result) {
     result.success(results);
   }
 
+  private void handleSendLocationUpdate(MethodCall call, Result result) {
+    Map<String, Object> locationData = call.arguments();
+    Location location = new Location("optimove");
+    location.setLatitude((Double) locationData.get("latitude"));
+    location.setLongitude((Double) locationData.get("longitude"));
+    location.setTime(((Double) locationData.get("time")).longValue());
```

Comment:
Think time is a `long`?

Reply:
Ah sorry see you read longValue. Is cast to Double first necessary?

Reply:
It is double in Flutter. The actual value should be the Java long yeah. The cast to Double is a protection (in case 17550000.0 was sent for some reason). The longValue() is a method of Double. The alternative would be to cast to long but again, the client couldnt potentially send .0